api/metadata_handler_medium_test.go

```diff
@@ -0,0 +1,133 @@
+// found in the LICENSE file.
+
+package api
+
+import (
+	"testing"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/web-platform-tests/wpt.fyi/api/query"
+	"github.com/web-platform-tests/wpt.fyi/shared"
+)
+
+
+func TestFilterMetadataHanlder_Success(t *testing.T) {
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		http.ServeFile(w, r, "../shared/metadata_testdata/gzip_testfile.tar.gz")
+	}
+	server := httptest.NewServer(http.HandlerFunc(handler))
+	defer server.Close()
+
+	r := httptest.NewRequest("GET", "/abd/api/metadata?product=chrome&product=safari", nil)
+	w := httptest.NewRecorder()
+	client := &http.Client{}
+
+	metadataHandler := MetadataHandler{nil, client, server.URL}
+	metadataHandler.ServeHTTP(w, r)
+
+	assert.Equal(t, http.StatusOK, w.Code)
+	res := w.Body.String()
+	assert.Equal(t, "[{\"test\":\"/IndexedDB/bindings-inject-key.html\",\"urls\":[\"bugs.chromium.org/p/chromium/issues/detail?id=934844\",\"\"]},{\"test\":\"/html/browsers/history/the-history-interface/007.html\",\"urls\":[\"bugs.chromium.org/p/chromium/issues/detail?id=592874\",\"\"]}]", res)
+}
+
+func TestFilterMetadataHanlder_MissingProducts(t *testing.T) {
+	r := httptest.NewRequest("GET", "/abd/api/metadata?", nil)
+	w := httptest.NewRecorder()
+	client := &http.Client{}
+
+	metadataHandler := MetadataHandler{nil, client,""}
+	metadataHandler.ServeHTTP(w, r)
+
+	assert.Equal(t, http.StatusBadRequest, w.Code)
+}
+
+func TestFilterMetadataSearchHandler_Success(t *testing.T) {
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		http.ServeFile(w, r, "../shared/metadata_testdata/gzip_testfile.tar.gz")
+	}
+	server := httptest.NewServer(http.HandlerFunc(handler))
+	defer server.Close()
+
+	body :=
+	`{
+		"run_ids": [0, 1, 2],
+		"query": {
+			"exists": [{
+				"link": "bugs.chromium.org"
+			}]
+		}
+	}`
+	bodyReader := strings.NewReader(body)
+	r := httptest.NewRequest("POST", "/abd/api/metadata?product=chrome&product=safari", bodyReader)
+	w := httptest.NewRecorder()
+	client := &http.Client{}
+
+	metadataHandler := MetadataSearchHandler{nil, client, server.URL}
+	metadataHandler.ServeHTTP(w, r)
+
+	assert.Equal(t, http.StatusOK, w.Code)
+	res := w.Body.String()
+	assert.Equal(t, "[{\"test\":\"/IndexedDB/bindings-inject-key.html\",\"urls\":[\"bugs.chromium.org/p/chromium/issues/detail?id=934844\",\"\"]},{\"test\":\"/html/browsers/history/the-history-interface/007.html\",\"urls\":[\"bugs.chromium.org/p/chromium/issues/detail?id=592874\",\"\"]}]", res)
+}
+
+func TestFilterMetadataSearchHandler_MissingProducts(t *testing.T) {
+	body :=
+	`{
+		"run_ids": [0, 1, 2],
+		"query": {
+			"exists": [{
+				"link": "bugs.chromium.org"
+			}]
+		}
+	}`
+	bodyReader := strings.NewReader(body)
+	r := httptest.NewRequest("GET", "/abd/api/metadata?", bodyReader)
+	w := httptest.NewRecorder()
+	client := &http.Client{}
+
+	metadataHandler := MetadataSearchHandler{nil, client,""}
```

Comment:
Nit:
```suggestion
	metadataHandler := MetadataSearchHandler{nil, client, ""}
```

Make sure you run `gofmt` automatically in your editor. (Unfortunately, we can't run `gofmt` on Travis because of annoying version mismatches.)